sympy/tensor/tests/test_tensor.py

```diff
@@ -1897,6 +1897,7 @@ def test_tensor_replacement():
     repl = {H(i,-j): [[1,2],[3,4]], L: diag(1, -1)}
     assert expr._extract_data(repl) == ([i, j], Array([[1, -2], [3, -4]]))
 
+    assert expr.replace_with_arrays(repl) == Array([[1, -2], [3, -4]])
```

Comment:
I guess that the discussing of the default value can be investigated by adding this test to the end of this sequence of tests. If it passes it should be safe to keep the default as `[]`. (Could be good to keep it there as well, just to confirm that it stays that way.)

Reply:
@oscargus Hello Oscar, I verified it with the example that you gave me. For your information, '[]' as default value can pass the test.
However, I actually use 'None' as default value for 'indices'. In my opinion, it can help avoir error when we pass 'None' as value to 'indices'(For unknown raison or by mistake), which is out of the ability of '[]' value solution. I have tested it as well. As you said, this would be the safest way.
Thank you for your advice!